types/default.go

```diff
@@ -23,6 +23,13 @@ var (
 )
 
 func getCachePath() string {
+	// os.HomeDir() fails when $HOME isn't set. Work around that behaviour
+	// here...
+	//
+	// https://github.com/jaypipes/pcidb/issues/38
+	if _, ok := os.LookupEnv("HOME"); !ok {
+		return ""
+	}
```

Comment:
IMO this defeats the whole object of using `os.UserHomeDir()`, since the point of it is that it's platform independent - different platforms use different env vars for home.

Since the *only* error which UserHomeDir can return is if the required env variable is not set, I think a better solution would be simply:

```
	hdir, err := os.UserHomeDir()
	if err != nil {
        return ""
    }
```

(that is, just remove the printing of the error)

Reply:
@candlerb great point. updated.